backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -120,6 +121,32 @@ public async Task<List<UserShortInfoDto>> GetFilteredUsers(UserFilterDto userFil
         return await AddCompatibility(filteredUsersList, filter.Compatibility);
     }
 
+    public async Task<UserShortInfoPaginationDto> GetFilteredUsersWithPagination(UserFilterWithNumberDto userFilter)
+    {
+        var originaUsers = await GetFilteredUsers(userFilter.Filter);
+
+        int usersCount = originaUsers.Count;
+
+        int remainingRange = usersCount - userFilter.PageNumber * AmountOfItemsOnPage;
+
+        int pagesCount = usersCount % AmountOfItemsOnPage == 0 ? usersCount / AmountOfItemsOnPage : usersCount / AmountOfItemsOnPage + 1;
+
+        int startIndex = userFilter.PageNumber * AmountOfItemsOnPage;
+        List<UserShortInfoDto> usersWithPagination;
+
+        if (remainingRange >= (AmountOfItemsOnPage))
+        {
+            usersWithPagination = originaUsers.GetRange(startIndex, AmountOfItemsOnPage);
+        }
+        else
+        {
+            usersWithPagination = originaUsers.GetRange(startIndex, remainingRange);
+        }
```

Comment:
this should be moved to the DB side

Reply:
I didn’t quite understand what needs to be done, there is no complex logic here, there is only one query to the database

Reply:
GetFilteredUsers returns a list of all users, and only after they you apply for pagination
But pagination should be applied on the DB side because we don't need to load all users to show only N of them